src/realm/query_expression.hpp

```diff
@@ -3714,175 +3724,98 @@ class SubQuery {
 };
 
 namespace aggregate_operations {
-template <typename T, typename Derived, typename R = T>
-class BaseAggregateOperation {
-    static_assert(realm::is_any<T, Int, Float, Double, Decimal128>::value,
-                  "Numeric aggregates can only be used with subcolumns of numeric types");
-
+template <typename T, typename Compare>
+class MinMaxAggregateOperator {
 public:
-    using ResultType = R;
-
     void accumulate(T value)
     {
-        m_count++;
-        m_result = Derived::apply(m_result, value);
-    }
-
-    void accumulate(util::Optional<T> value)
-    {
-        if (value) {
-            m_count++;
-            m_result = Derived::apply(m_result, *value);
+        if (!m_result || Compare()(value, *m_result)) {
+            m_result = value;
```

Comment:
Does this work with a nullable Decimal128 list where the first item is null? I can't remember if we have a test checking min/max of that case.

Reply:
Ah, I bet it is wrong. If I can't find any tests for that I'll add some.

Reply:
I ended up writing a fairly excessive test and it turned out that floating point nulls weren't handled correctly in the old code either. Luckily it turned out to be a very simple fix.

Reply:
I'm not surprised about bugs lurking in this code, thanks for constructing such a nice comprehensive test! 
I'm still a little suspicious about the above check for the non-nullable float list where the first element is NaN - does that case work?

Reply:
NaNs currently just poison the whole thing even if it's a nullable list. I think the best option is probably to ignore them for the sake of aggregate operations just like with nulls?

Reply:
Ok strange, I thought we had coverage for this but I guess not. But yes, I would expect them to be treated as null, and not aggregated. At least that seems the most reasonable behaviour to me.

Reply:
I've updated it to test NaNs in the same new test and ignore them when aggregating.